langkit/regexes.py

```diff
@@ -73,22 +71,28 @@ def get_regex_groups(self):
 
 def has_patterns(text: str) -> Optional[str]:
     regex_groups = pattern_loader.get_regex_groups()
-    patterns_info = None
     if regex_groups:
         for group in regex_groups:
             for expression in group["expressions"]:
                 if expression.search(text):
-                    patterns_info = group["name"]
                     return group["name"]
-    return patterns_info
+    return None
 
 
 if pattern_loader.get_regex_groups() is not None:
     register_metric_udf(col_type=String, type_mapper=AllString())(has_patterns)
 
 
-def init(pattern_file_path: Optional[str] = None):
+def init(
+    pattern_file_path: Optional[str] = None, lang_config: Optional[LangKitConfig] = None
+):
+    if lang_config is None:
+        lang_config = LangKitConfig()
     if pattern_file_path:
         lang_config.pattern_file_path = pattern_file_path
         pattern_loader.set_config(lang_config)
         pattern_loader.update_patterns()
+    else:
+        lang_config = LangKitConfig()
```

Comment:
Line 96 looks fishy in context of line 90?

Reply:
good catch, don't want to overwrite passed in config (or extra instantiation). Fixed!